### DISCUSSION TITLE
How to set absolute position of widget?

### DISCUSSION THREAD

**Discussion by Stingaaa** [cogentcore/core#1302] [Q&A] [answered]
I tried to set the position of my widgets in various ways (Frame => NoLayout & styler => s.Pos.Set or s.Margin.Set; etc.) but everything does not work as I would ike it to.
Whilst s.Margin.Set works with Frame => NoLayout it only displays the first widget, all others automatically have the "Invisible" State set.
Currently I don't see any option I didnt try, could someone please give me an example/ solution how I can use an "absolute" layouting style?

**Comment by kkoreilly** [comment, marked as answer]
This is a duplicate of #1186, which we are planning to implement within a couple of days (`s.Pos` is the solution and we just have to finish implementing it).

**Reply by Stingaaa** [reply]
Sorry for wasting your time, I looked at quite a few discussions but it seems like I overlooked #1186.
Thank you for the reply.

**Reply by kkoreilly** [reply]
No worries! I am always happy to answer questions.

**Reply by Stingaaa** [reply]
I have tried a few different things to find a workaround until the absolute layout feature is implemented, but I am not able to achieve my ideal layout which should look as follows:
 ![Bild_2024-11-13_225152577](https://github.com/user-attachments/assets/55dca273-d4f0-4e88-a8d8-d23bea5a3c2a)
My main problems are that I do not know, how I can put all those widgets into one view without exceeding the max width and/or height and making them overlap.
Is there a way to achieve such a layout?
If yes, would you mind showing me which parameters I have to use to achieve this?

**Reply by kkoreilly** [reply]
As of #1303 the absolute layout `s.Pos` feature is implemented, so you can try that out now. Thank you for the helpful diagram of your use case; that layout is definitely possible, and I will give you guidance on it soon.

**Reply by Stingaaa** [reply]
Thank you very much for the reply, I don't want to waste your time unneccesarily, so I will look into it and will hopefully fix my issues.
If I am not able to, I will gladly accept your help, but as I have mentioned before, I believe you have better things to do, than helping some lost soul, who didn't even try to read through your new implementations.

**Reply by Stingaaa** [reply]
One little detail I noticed in your recent merge, I am not sure is intended, is that you print the position set in core/layout.go line 1183.

**Reply by kkoreilly** [reply]
Thank you for pointing that out! It is indeed unintended; we will fix that now.

**Reply by kkoreilly** [reply]
As of #1305, the print statement is removed. We also renamed `NoLayout` to `Custom` to better represent the functionality.

**Reply by Stingaaa** [reply]
Amazing, thank you for the reply.
